orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -115,6 +115,14 @@ def forward(ctx, *inputs):
                 # converted to a tensor filled with zeros prior to calling backward.
                 # Save shape, device and type info to ctx for materializing tensor in backward if output grad is None.
                 ctx.set_materialize_grads(False)
+
+                # Mark the outputs tensors needed in backward computation
+                # ORT is NOT relying on save_for_backward() to actually save the tensor, 
+                # as this tensor is also kept in ORT's PartialGraphState
+                # This call is to invoke pytorch's version check to detect the potential inplace corruption
+                for idx in self._graph_info.module_output_indice_requires_save_for_backward:
```

Comment:
do we understand why the corruption happen?

Reply:
let's say we have output = model(input) 

If user do some inplace operation on the `output` like this : 
output +=1   
output.add_(1)   
relu= nn.Relu(**inplace=True**)    
y = relu(output)   

and if the output is also needed for backward computation, we will hit such corruption . 

Luckily, pytorch is already handling such case even if model is an nn.module. 
So I am also making ORTModule to follow the nn.module behavior.

Reply:
got it. any change to the tensor between a forward and backward would have this effect.
This change will help pytorch detect such change and an exception would be raised to warn the user, is that it?

It is not clear what is the intended behavior after the fix

Reply:
yes, this will help pytorch detect such change and error out to warn the user.